src/ui/common/src/components/integrations/cards/text.tsx

```diff
@@ -30,24 +41,52 @@ export const ResourceCardText: React.FC<ResourceCardTextProps> = ({
   };
   const labelWidthNum = Math.max(...labels.map(useLabelWidth));
 
-  return (
-    <Box sx={{ display: 'flex', flexDirection: 'column' }}>
-      {labels.map((_, index) => (
-        <Box key={index} sx={{ display: 'flex', flexWrap: 'nowrap' }}>
-          <TruncatedText
-            variant="body2"
-            sx={{ fontWeight: 300, width: `${labelWidthNum}px` }}
-          >
-            {labels[index]}
-          </TruncatedText>
-          <TruncatedText
-            variant="body2"
-            sx={{ width: `calc(100% - ${labelWidthNum}px)` }}
-          >
-            {values[index]}
-          </TruncatedText>
-        </Box>
-      ))}
-    </Box>
-  );
+  // Chunk the fields into their respective columns.
+  const chunkedLabels = chunkList(labels, maxNumRows);
+  const chunkedValues = chunkList(values, maxNumRows);
+
+  const labelAndValueColumn = (labelsForColumn: string[], colIndex: number) => {
```

Comment:
@kenxu95, this is a fine way to do it for now, but in the future, you can simplify this significantly by using the [`Grid`](https://mui.com/material-ui/react-grid/) component from MUI. It will do a dynamic/reactive layout for you, and it'll allow you to declaratively specify the number of rows/columns/etc.

Reply:
Gotcha - I'll make a task for this Grid refactor in general since I think some other places need to use it too.